flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -408,12 +408,15 @@ public void runFetcher() throws Exception {
 	 */
 	public void shutdownFetcher() {
 		running = false;
-		mainThread.interrupt(); // the main thread may be sleeping for the discovery interval
-
+		if (mainThread != null) {
+			mainThread.interrupt(); // the main thread may be sleeping for the discovery interval
+		}
 		if (LOG.isInfoEnabled()) {
 			LOG.info("Shutting down the shard consumer threads of subtask {} ...", indexOfThisConsumerSubtask);
 		}
-		shardConsumersExecutor.shutdownNow();
+		if (shardConsumersExecutor != null) {
```

Comment:
Actually I think this is a redundant `null` check, because `shardConsumersExecutor` is final.
It should never be null, so this null check might actually be confusing to other readers.